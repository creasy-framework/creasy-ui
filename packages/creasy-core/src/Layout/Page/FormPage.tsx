import React, { FunctionComponent, ReactElement } from 'react';
import { Scrollable } from '../../Interactive';
import PageContent from './PageContent';
import useStickyHeader from './useStickyHeader';
import BasePage, { BasePageProps } from "./BasePage";

interface Props extends BasePageProps {
  contentHeaderStickyStyle: 'none' | 'always' | 'scrollUpOnly';
  contentContainerStyle: 'none' | 'card';
  primaryContentHeader?: ReactElement;
  secondaryContentHeader?: ReactElement;
  contentFooter?: ReactElement;
  spinner?: ReactElement;
  isProcessing?: boolean;
}

const FormPage: FunctionComponent<Props> = ({
  pageHeader,
  pageFooter,
  primaryContentHeader,
  secondaryContentHeader,
  contentFooter,
  children,
  spinner,
  isProcessing,
  className,
  contentHeaderStickyStyle = 'none',
  contentContainerStyle = 'none'
}) =>  {
  const [contentHeaderRef, onScroll] = useStickyHeader(contentHeaderStickyStyle);
  return (
    <BasePage className={className} pageHeader={pageHeader} pageFooter={pageFooter} isProcessing={isProcessing} spinner={spinner}>
      <Scrollable onScroll={onScroll as any}>
        <PageContent
          headerRef={contentHeaderRef}
          primaryHeader={primaryContentHeader}
          secondaryHeader={secondaryContentHeader}
          footer={contentFooter}
          headerStickyStyle={contentHeaderStickyStyle}
          contentStyle={contentContainerStyle}
        >
          {children}
        </PageContent>
      </Scrollable>
    </BasePage>
  );
}

export default FormPage;
