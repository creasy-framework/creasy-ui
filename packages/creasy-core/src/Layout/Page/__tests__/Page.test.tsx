import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormPage from '../FormPage';

describe('FormPage', () => {
  const Header = () => <>Header</>;
  const Content = () => <>Content</>;
  const Footer = () => <>Footer</>;

  it('should render properly', () => {
    const wrapper = shallow(
      <FormPage pageHeader={<Header/>} pageFooter={<Footer/>}>
        <Content/>
      </FormPage>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
