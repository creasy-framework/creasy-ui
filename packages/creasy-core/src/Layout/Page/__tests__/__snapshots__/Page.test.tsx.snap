// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPage should render properly 1`] = `
<BasePage
  pageFooter={<Footer />}
  pageHeader={<Header />}
>
  <Scrollable
    onScroll={[Function]}
  >
    <PageContent
      contentStyle="none"
      headerRef={
        Object {
          "current": undefined,
        }
      }
      headerStickyStyle="none"
    >
      <Content />
    </PageContent>
  </Scrollable>
</BasePage>
`;
